conductr_cli/conductr_restore.py

```diff
@@ -0,0 +1,134 @@
+import json
+import os
+import shutil
+import tempfile
+
+import io
+import copy
+import logging
+
+import sys
+
+from io import BytesIO
+
+from conductr_cli import control_protocol, bundle_utils, validation, conduct_load, bundle_scale
+from conductr_cli.bundle_core_info import BundleCoreInfo
+from conductr_cli.exceptions import ConductRestoreError
+
+
+@validation.handle_connection_error
+@validation.handle_http_error
+@validation.handle_wait_timeout_error
+@validation.handle_conductr_restore_error
+def restore(args):
+    log = logging.getLogger(__name__)
+
+    isatty = os.isatty(sys.stdin.fileno())
+    if args.backup is '-' and isatty:
+        raise ConductRestoreError('Specify a backup file. Cannot read from stdin(-)')
+
+    destination_bundles = control_protocol.get_bundles(args)
+    destination_bundles_info = BundleCoreInfo.from_bundles(destination_bundles)
+
+    restore_directory = unpack_backup(args.backup)
+    bundle_json_file = open(os.path.join(restore_directory, 'bundles.json'), 'r')
+    bundles_json = bundle_json_file.read()
+    bundle_json_file.close()
+
+    bundles_info = sorted(BundleCoreInfo.from_bundles(json.loads(bundles_json)), key=lambda b: b.start_time)
+
+    restore_errors = []
+    for bundle_info in bundles_info:
```

Comment:
Eventually I would like to see both loading and scaling for a bundle together make that a unit of work and then make all of that work async.
I think python allows that now, so we should look into this separately. Would be good for large restores as well. Same for backup as well.